static/js/generic_checkbox.html

```diff
@@ -0,0 +1,304 @@
+<html>
+
+<style>
+
+.bar rect {
+  fill: steelblue;
+  shape-rendering: crispEdges;
+}
+
+.bar text {
+  fill: #fff;
+}
+
+.axis path, .axis line {
+  fill: none;
+  stroke: #000;
+  shape-rendering: crispEdges;
+}
+
+</style>
+
+<head>
+	<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
+	<script src="http://52.50.241.19/course-content/2017_N/static/js/server_interaction.js" type="text/javascript"></script>
+	<script src="http://d3js.org/d3.v2.min.js?2.10.0"></script>
+	<script>
+
+		function getUrlVars() {
+		  var vars = {};
+		  var url = document.URL;
+		  url = url.split('%24').join('&');
+		  url = url.split('%3D').join('=');
+
+		  var parts = url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+			vars[key] = value;
+		  });
+		  return vars;
+		}
+
+		/*
+			This is a global state dictionary. We define this for convenience when working
+			with edX's state-saving paradigm. Any data that is required for operation to
+			seamlessly persist across page refreshes, or that EdX needs to know about, must
+			be placed in here.
+		*/
+		state = {
+			//is the student's answer correct? (possibly a misnomer... eh)
+				//required for grading the problem
+			'correct_answer' : false,
+			//which choice is the answer?
+			'answer' : -1,
+			//list of user inputs
+				//required to repopulate the form so users don't shit bricks
+			'inputs' : [],
+			//attempts at the problem (how many times getGrade() has been called)
+				//required for our adaptive hints
+			'attempts' : 0,
+			//number of choices
+			'choices' : 0
+		};
+
+		//global variable that is set once the form is displayed.
+		//If this is false we can't access the form's document elements,
+		//because they haven't yet been created.
+		displayed = false;
+
+		//url variables
+		vars = {};
+
+		//just stupid rewrapping the url vars into a global variable
```

Comment:
I know javascript is far from perfect, but there's no need to complain about it in the comments ;)